FastSimulation/SimplifiedGeometryPropagator/plugins/NuclearInteraction.cc

```diff
@@ -200,8 +202,8 @@ namespace fastsim
     // TODO: Is this correct?
     // Thread safety
     static std::once_flag initializeOnce;
-    [[cms::thread_guard("initializeOnce")]] std::vector< std::vector<double> > NuclearInteraction::theRatiosMap;
-    [[cms::thread_guard("initializeOnce")]] std::map<int, int> NuclearInteraction::theIDMap;
+    CMS_THREAD_GUARD(initializeOnce) std::vector< std::vector<double> > NuclearInteraction::theRatiosMap;
```

Comment:
@Dr15Jones do you see what is different here wrt to the case of NuclearInteractionFTF ? I might simply restore the previous syntax removing the once_flag declaration, but I do not understand what is wrong here...

Reply:
indeed the other possible solution works in case

Reply:
` CMS_THREAD_GUARD(initializeOnce)` does not actually use the variable `initializeOnce`, it just creates the string `"initializeOnce"`. As far as I can tell, the variable is not actually used.

Reply:
The varaibles `theRatiosMap` and `theIDMap` are not handled in a thread-safe way.

Reply:
@Dr15Jones ok, in the case of NuclearInteractionFTF initializeOnce is then used by std::call_once, that is the difference